src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -363,28 +387,34 @@ def remove_properties(instance, argument_values):
     if not list_index:
         _find_property(instance, list_attribute_path)
         parent_to_remove_from = _find_property(instance, list_attribute_path[:-1])
-        del parent_to_remove_from[list_attribute_path[-1]]
+        if isinstance(parent_to_remove_from, dict):
+            del parent_to_remove_from[list_attribute_path[-1]]
+        elif hasattr(parent_to_remove_from, make_snake_case(list_attribute_path[-1])):
+            setattr(parent_to_remove_from, make_snake_case(list_attribute_path[-1]), None)
+        else:
+            raise ValueError
```

Comment:
>raise ValueError [](start = 12, length = 16)

we should probably include an error text here.  Also, is this something the user will see?  If so, we should use a CLIError